# Thinking mode: cracking the dataset TFMs couldn't

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/Innixma/kdd2026_tutorial_materials/blob/main/notebooks/05_thinking_mode.ipynb)
[![GitHub Repo](https://img.shields.io/badge/GitHub-Repo-181717?logo=github)](https://github.com/Innixma/kdd2026_tutorial_materials)
[![Tutorial Website](https://img.shields.io/badge/Tutorial-Website-0a7aca?logo=googlechrome&logoColor=white)](https://kdd26-automl-hands-on.github.io/)

**Taming Structured Data Foundation Models with AutoML — KDD 2026 hands-on tutorial**

*Amazon employee access* is a famous holdout of the pre-TFM era: ~33k rows whose nine
features are all high-cardinality categorical codes (resource ids, manager ids, role
codes). It is CatBoost's home turf — and on the [TabArena](https://tabarena.ai)
leaderboard it is one of the very few datasets where **every** tabular foundation model
loses to well-tuned gradient boosting.

This notebook first reproduces that struggle live (CatBoost vs TabICLv2 and TabPFN-3,
with EXAONE's and TabFM's benchmark scores quoted alongside), then runs
[TabPFN-3](https://priorlabs.ai/technical-reports/tabpfn-3)'s
[thinking mode](https://docs.priorlabs.ai/capabilities/thinking-mode) on the same split.

> **Runtime**: ~10-15 minutes on a Colab T4; the thinking-mode fit runs on the TabPFN API
> (about 5 minutes at high effort) rather than the local GPU.

## Setup

The install is skipped outside Colab so it never overwrites a locally managed environment.

In [1]:
import importlib.util

IN_COLAB = importlib.util.find_spec("google.colab") is not None
if IN_COLAB:
    !command -v uv >/dev/null || pip install -q uv
    !uv pip install -q --python {__import__('sys').executable} "autogluon.tabular[tabarena]" openml tabpfn-client

    # The install may replace Colab's preinstalled numpy; the copy already loaded in this
    # kernel then no longer matches the files on disk and imports break. When that happens,
    # restart the runtime once (continue from the next cell after it reconnects).
    import importlib.metadata
    import numpy
    if importlib.metadata.version("numpy") != numpy.__version__:
        print("numpy changed -- restarting the Colab runtime; re-run FROM THE NEXT CELL when it reconnects.")
        import os
        os.kill(os.getpid(), 9)

## The dataset

Split 0 of the official benchmark task, as everywhere in this tutorial.

In [2]:
import getpass
import os

import openml
from autogluon.tabular import TabularDataset, TabularPredictor

# --- Prior Labs access token (unlocks the gated TabPFN checkpoints and the API) ---
# 1. Sign up / log in at https://ux.priorlabs.ai
# 2. Accept the license at https://ux.priorlabs.ai/account/licenses
# 3. Copy your access token from https://ux.priorlabs.ai/account
# Tip: save it as a Colab secret named TABPFN_TOKEN to skip the prompt next time.
tabpfn_token = os.environ.get("TABPFN_TOKEN")
if not tabpfn_token:
    try:
        from google.colab import userdata
        tabpfn_token = userdata.get("TABPFN_TOKEN")
    except Exception:
        pass
while not tabpfn_token:
    tabpfn_token = getpass.getpass("Paste your TABPFN_TOKEN and press Enter: ").strip()
os.environ["TABPFN_TOKEN"] = tabpfn_token

task = openml.tasks.get_task(363613)  # Amazon_employee_access
X, y = task.get_X_and_y(dataset_format="dataframe")
train_idx, test_idx = task.get_train_test_split_indices(repeat=0, fold=0)

label = y.name
full_data = X.copy()
full_data[label] = y
train_data = TabularDataset(full_data.iloc[train_idx].reset_index(drop=True))
test_data = TabularDataset(full_data.iloc[test_idx].reset_index(drop=True))
print(f"train: {train_data.shape}, test: {test_data.shape}")
train_data.head(3)

train: (21846, 10), test: (10923, 10)


,RESOURCE,MGR_ID,ROLE_ROLLUP_1,ROLE_ROLLUP_2,ROLE_DEPTNAME,ROLE_TITLE,ROLE_FAMILY_DESC,ROLE_FAMILY,ROLE_CODE,ResourceApproved
0,77465,21698,118192,118193,117895,118194,118195,117887,118196,Yes
1,5764,2858,118219,118220,118344,127847,128352,118347,127848,Yes
2,20299,26345,117929,117930,117920,124313,131967,120134,124315,Yes


## Round 1 — the TFM struggle, live

CatBoost against the foundation models, single fits on identical data. Two more TFMs
(EXAONE-Tabular and TabFM) are commented out — extra install and a very large checkpoint
download respectively; their benchmark scores appear in the closing comparison.

In [3]:
# tabarena's benchmark wrappers drop into the same dict as classes. Install them with:
#   pip install "tabarena[exaone_tabular] @ git+https://github.com/autogluon/tabarena.git#subdirectory=packages/tabarena"
# from tabarena.models.exaone_tabular.model import EXAONETabularModel
# from tabarena.models.tabfm.model import TabFMModel  # ~13GB checkpoint download

predictor = TabularPredictor(label=label, eval_metric="roc_auc").fit(
    train_data,
    hyperparameters={
        "CAT": {},
        "TABICL": {},           # TabICLv2
        "TABPFN-3": {},         # noncommercial weights; via the access token above
        # EXAONETabularModel: {},
        # TabFMModel: {"n_estimators": 1},
    },
    num_gpus=1,
)
predictor.leaderboard(test_data)

No path specified. Models will be saved in: "AutogluonModels/ag-20260809_062913"


Verbosity: 2 (Standard Logging)


=================== System Info ===================
AutoGluon Version:  1.6.1.dev0
Python Version:     3.11.15
Operating System:   Linux
Platform Machine:   x86_64
Platform Version:   #22~24.04.1-Ubuntu SMP Sat Nov 22 06:23:18 UTC 2025
CPU Count:          192
Pytorch Version:    2.13.0+cu130
CUDA Version:       13.0
GPU Memory:         GPU 0: 94.97/94.97 GB
Total GPU Memory:   Free: 94.97 GB, Allocated: 0.00 GB, Total: 94.97 GB
GPU Count:          1
Memory Avail:       1387.32 GB / 1417.32 GB (97.9%)
Disk Space Avail:   1501.97 GB / 9984.00 GB (15.0%)


No presets specified! To achieve strong results with AutoGluon, it is recommended to use the available presets. Defaulting to `'medium'`...
	Recommended Presets (For more details refer to https://auto.gluon.ai/stable/tutorials/tabular/tabular-essentials.html#presets):
	presets='extreme'  : Use this if you have a GPU. The go-to preset for best results, and the one to use for benchmark comparisons. New in v1.6: far better than 'best' on datasets <100000 samples by using Tabular Foundation Models (TFMs) meta-learned on https://tabarena.ai: Nori, TabICLv2, and TabDPT-Turbo. Every model is free for commercial use. Requires `pip install autogluon.tabular[tabarena]`.
	presets='noncommercial': New in v1.6: 'extreme' plus TabPFN-3, a frontier tabular foundation model created by Prior Labs. Stronger still, but commercial use requires a TabPFN-3 license: https://docs.priorlabs.ai/models#tabpfn-model-license
	presets='best'     : Use this if you do not have a GPU. Maximize accuracy. Use in competi

Beginning AutoGluon training ...


AutoGluon will save models to "/home/nick_priorlabs_ai/workspace_tabpfn_plus/code/kdd2026_tutorial_materials/notebooks/AutogluonModels/ag-20260809_062913"


Train Data Rows:    21846


Train Data Columns: 9


Label Column:       ResourceApproved


AutoGluon infers your prediction problem is: 'binary' (because only two unique label-values observed).


	2 unique label values:  ['Yes', 'No']


	If 'binary' is not the correct problem_type, please manually specify the problem_type parameter during Predictor init (You may specify problem_type as one of: ['binary', 'multiclass', 'regression', 'quantile'])


Problem Type:       binary


Preprocessing data...


Selected class <--> label mapping:  class 1 = Yes, class 0 = No


	Note: For your binary classification, AutoGluon arbitrarily selected which label-value represents positive (Yes) vs negative (No) class.
	To explicitly set the positive_class, either rename classes to 1 and 0, or specify positive_class in Predictor init.


Using Feature Generators to preprocess the data ...


Fitting AutoMLPipelineFeatureGenerator...


	Available Memory:                    1420635.02 MB


	Train Data (Original)  Memory Usage: 1.28 MB (0.0% of available memory)


	Inferring data type of each feature based on column values. Set feature_metadata_in to manually specify special dtypes of the features.


	Stage 1 Generators:


		Fitting AsTypeFeatureGenerator...


	Stage 2 Generators:


		Fitting FillNaFeatureGenerator...


	Stage 3 Generators:


		Fitting CategoryFeatureGenerator...


			Fitting CategoryMemoryMinimizeFeatureGenerator...


	Stage 4 Generators:


		Fitting DropUniqueFeatureGenerator...


	Stage 5 Generators:


		Fitting DropDuplicatesFeatureGenerator...


	Unused Original Features (Count: 1): ['ROLE_CODE']


		These features were not used to generate any of the output features. Add a feature generator compatible with these features to utilize them.


		Features can also be unused if they carry very little information, such as being categorical but having almost entirely unique values or being duplicates of other features.


		These features do not need to be present at inference time.


		('category', []) : 1 | ['ROLE_CODE']


	Types of features in original data (raw dtype, special dtypes):


		('category', []) : 8 | ['RESOURCE', 'MGR_ID', 'ROLE_ROLLUP_1', 'ROLE_ROLLUP_2', 'ROLE_DEPTNAME', ...]


	Types of features in processed data (raw dtype, special dtypes):


		('category', []) : 8 | ['RESOURCE', 'MGR_ID', 'ROLE_ROLLUP_1', 'ROLE_ROLLUP_2', 'ROLE_DEPTNAME', ...]


	0.1s = Fit runtime


	8 features in original data used to generate 8 features in processed data.


	Train Data (Processed) Memory Usage: 0.30 MB (0.0% of available memory)


Data preprocessing and feature engineering runtime = 0.08s ...


AutoGluon will gauge predictive performance using evaluation metric: 'roc_auc'


	This metric expects predicted probabilities rather than predicted class labels, so you'll need to use predict_proba() instead of predict()


	To change this, specify the eval_metric parameter of Predictor()


Automatically generating train/validation split with holdout_frac=0.1, Train Rows: 19661, Val Rows: 2185


User-specified model hyperparameters to be fit:
{
	'CAT': [{}],
	'TABICL': [{}],
	'TABPFN-3': [{}],
}


Fitting 3 L1 models, fit_strategy="sequential" ...


Fitting model: CatBoost ...


	Fitting with cpus=192, gpus=1


	Training CatBoost with GPU, note that this may negatively impact model quality compared to CPU training.


	0.8724	 = Validation score   (roc_auc)


	22.2s	 = Training   runtime


	0.02s	 = Validation runtime


Fitting model: TabICL ...


	Fitting with cpus=192, gpus=1, mem=4.1/1388.1 GB


/home/nick_priorlabs_ai/workspace_tabpfn_plus/venv/.venv/lib/python3.11/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm


	0.8466	 = Validation score   (roc_auc)


	1.35s	 = Training   runtime


	0.87s	 = Validation runtime


Fitting model: TabPFN-3 ...


	Fitting with cpus=192, gpus=1, mem=2.6/1387.3 GB


/home/nick_priorlabs_ai/workspace_tabpfn_plus/venv/.venv/lib/python3.11/site-packages/pandas/core/arrays/base.py:568: RuntimeWarning: invalid value encountered in cast
  result = np.asarray(self, dtype=dtype)
/home/nick_priorlabs_ai/workspace_tabpfn_plus/venv/.venv/lib/python3.11/site-packages/pandas/core/arrays/base.py:568: RuntimeWarning: invalid value encountered in cast
  result = np.asarray(self, dtype=dtype)


	0.8469	 = Validation score   (roc_auc)


	1.19s	 = Training   runtime


	1.69s	 = Validation runtime


Fitting model: WeightedEnsemble_L2 ...


	Fitting 1 model on all data | Fitting with cpus=192, gpus=1, mem=0.0/1386.1 GB


	Ensemble Weights: {'CatBoost': 0.667, 'TabPFN-3': 0.333}


	0.8784	 = Validation score   (roc_auc)


	0.02s	 = Training   runtime


	0.0s	 = Validation runtime


AutoGluon training complete, total runtime = 28.92s ... Best model: WeightedEnsemble_L2 | Estimated inference throughput: 1281.0 rows/s (2185 batch size)


TabularPredictor saved. To load, use: predictor = TabularPredictor.load("/home/nick_priorlabs_ai/workspace_tabpfn_plus/code/kdd2026_tutorial_materials/notebooks/AutogluonModels/ag-20260809_062913")


/home/nick_priorlabs_ai/workspace_tabpfn_plus/venv/.venv/lib/python3.11/site-packages/pandas/core/arrays/base.py:568: RuntimeWarning: invalid value encountered in cast
  result = np.asarray(self, dtype=dtype)


,model,score_test,score_val,eval_metric,pred_time_test,pred_time_val,fit_time,pred_time_test_marginal,pred_time_val_marginal,fit_time_marginal,stack_level,can_infer,fit_order
0,WeightedEnsemble_L2,0.856018,0.878387,roc_auc,2.517168,1.705689,23.407791,0.003270,0.000371,0.016575,2,True,4
1,CatBoost,0.848068,0.872371,roc_auc,0.157109,0.016136,22.196251,0.157109,0.016136,22.196251,1,True,1
2,TabICL,0.838881,0.846560,roc_auc,1.276549,0.869720,1.352124,1.276549,0.869720,1.352124,1,True,2
3,TabPFN-3,0.832236,0.846943,roc_auc,2.356789,1.689182,1.194965,2.356789,1.689182,1.194965,1,True,3


The pattern the benchmark shows across nine splits holds in one glance: **CatBoost on top,
every foundation model behind it**. Wide, high-cardinality categorical spaces are the
corner of tabular learning where gradient boosting still rules and in-context learning has
struggled.

## Round 2 — thinking mode

Same split, one change: TabPFN-3 through the API with
[thinking mode](https://docs.priorlabs.ai/capabilities/thinking-mode) enabled. It spends
substantially more compute at fit time (about five minutes here), steered toward the metric
you declare.

In [4]:
import time

import tabpfn_client
from sklearn.metrics import roc_auc_score
from tabpfn_client import TabPFNClassifier

tabpfn_client.set_access_token(os.environ["TABPFN_TOKEN"])

X_train, y_train = X.iloc[train_idx], y.iloc[train_idx]
X_test, y_test = X.iloc[test_idx], y.iloc[test_idx]
y_train_bin = (y_train == y_train.cat.categories[1]).astype(int) if hasattr(y_train, "cat") else y_train
y_test_bin = (y_test == y_test.cat.categories[1]).astype(int) if hasattr(y_test, "cat") else y_test

clf = TabPFNClassifier(
    thinking_mode=True,
    thinking_effort="high",
    thinking_metric="roc_auc",
)
t0 = time.time()
clf.fit(X_train, y_train_bin)
fit_s = time.time() - t0
proba = clf.predict_proba(X_test)[:, 1]
print(f"TabPFN-3 (thinking): AUC = {roc_auc_score(y_test_bin, proba):.4f}   (fit {fit_s:.0f}s)")

00:00 Fitting... -

The provided train set hashes match previously uploaded train sets.


00:00 Fitting... \

00:00 Fitting... |

00:00 Fitting... /

00:00 Fitting... -

00:01 Fitting... \

00:01 Fitting... |

00:01 Fitting... /

00:01 Fitting... -

00:01 Fitting... \

00:02 Fitting... |

00:02 Fitting... /

00:02 Fitting... -

00:02 Fitting... \

00:02 Fitting... |

00:03 Fitting... /

00:03 Fitting... -

00:03 Fitting... \

00:03 Fitting... |

00:03 Fitting... /

00:04 Fitting... -

00:04 Fitting... \

00:04 Fitting... |

00:04 Fitting... /

00:04 Fitting... -

00:05 Fitting... \

00:05 Fitting... |

00:05 Fitting... /

00:05 Fitting... -

00:05 Fitting... \

00:06 Fitting... |

00:06 Fitting... /

00:06 Fitting... -

00:06 Fitting... \

00:06 Fitting... |

00:07 Fitting... /

00:07 Fitting... -

00:07 Fitting... \

00:07 Fitting... |

00:07 Fitting... /

00:08 Fitting... -

00:08 Fitting... \

00:08 Fitting... |

00:08 Fitting... /

00:08 Fitting... -

00:09 Fitting... \

00:09 Fitting... |

00:09 Fitting... /

00:09 Fitting... -

00:09 Fitting... \

00:10 Fitting... |

00:10 Fitting... /

00:10 Fitting... -

00:10 Fitting... \

00:10 Fitting... |

00:11 Fitting... /

00:11 Fitting... -

00:11 Fitting... \

00:11 Fitting... |

00:11 Fitting... /

00:12 Fitting... -

00:12 Fitting... \

00:12 Fitting... |

00:12 Fitting... /

00:12 Fitting... -

00:13 Fitting... \

00:13 Fitting... |

00:13 Fitting... /

00:13 Fitting... -

00:13 Fitting... \

00:14 Fitting... |

00:14 Fitting... /

00:14 Fitting... -

00:14 Fitting... \

00:14 Fitting... |

00:15 Fitting... /

00:15 Fitting... -

00:15 Fitting... \

00:15 Fitting... |

00:15 Fitting... /

00:16 Fitting... -

00:16 Fitting... \

00:16 Fitting... |

00:16 Fitting... /

00:16 Fitting... -

00:17 Fitting... \

00:17 Fitting... |

00:17 Fitting... /

00:17 Fitting... -

00:17 Fitting... \

00:18 Fitting... |

00:18 Fitting... /

00:18 Fitting... -

00:18 Fitting... \

00:18 Fitting... |

00:19 Fitting... /

00:19 Fitting... -

00:19 Fitting... \

00:19 Fitting... |

00:19 Fitting... /

00:20 Fitting... -

00:20 Fitting... \

00:20 Fitting... |

00:20 Fitting... /

00:20 Fitting... -

00:21 Fitting... \

00:21 Fitting... |

00:21 Fitting... /

00:21 Fitting... -

00:21 Fitting... \

00:22 Fitting... |

00:22 Fitting... /

00:22 Fitting... -

00:22 Fitting... \

00:22 Fitting... |

00:23 Fitting... /

00:23 Fitting... -

00:23 Fitting... \

00:23 Fitting... |

00:23 Fitting... /

00:24 Fitting... -

00:24 Fitting... \

00:24 Fitting... |

00:24 Fitting... /

00:24 Fitting... -

00:25 Fitting... \

00:25 Fitting... |

00:25 Fitting... /

00:25 Fitting... -

00:25 Fitting... \

00:26 Fitting... |

00:26 Fitting... /

00:26 Fitting... -

00:26 Fitting... \

00:26 Fitting... |

00:27 Fitting... /

00:27 Fitting... -

00:27 Fitting... \

00:27 Fitting... |

00:27 Fitting... /

00:28 Fitting... -

00:28 Fitting... \

00:28 Fitting... |

00:28 Fitting... /

00:28 Fitting... -

00:29 Fitting... \

00:29 Fitting... |

00:29 Fitting... /

00:29 Fitting... -

00:29 Fitting... \

00:30 Fitting... |

00:30 Fitting... /

00:30 Fitting... -

00:30 Fitting... \

00:30 Fitting... |

00:31 Fitting... /

00:31 Fitting... -

00:31 Fitting... \

00:31 Fitting... |

00:31 Fitting... /

00:32 Fitting... -

00:32 Fitting... \

00:32 Fitting... |

00:32 Fitting... /

00:32 Fitting... -

00:33 Fitting... \

00:33 Fitting... |

00:33 Fitting... /

00:33 Fitting... -

00:33 Fitting... \

00:34 Fitting... |

00:34 Fitting... /

00:34 Fitting... -

00:34 Fitting... \

00:34 Fitting... |

00:35 Fitting... /

00:35 Fitting... -

00:35 Fitting... \

00:35 Fitting... |

00:35 Fitting... /

00:36 Fitting... -

00:36 Fitting... \

00:36 Fitting... |

00:36 Fitting... /

00:36 Fitting... -

00:37 Fitting... \

00:37 Fitting... |

00:37 Fitting... /

00:37 Fitting... -

00:37 Fitting... \

00:38 Fitting... |

00:38 Fitting... /

00:38 Fitting... -

00:38 Fitting... \

00:38 Fitting... |

00:39 Fitting... /

00:39 Fitting... -

00:39 Fitting... \

00:39 Fitting... |

00:39 Fitting... /

00:40 Fitting... -

00:40 Fitting... \

00:40 Fitting... |

00:40 Fitting... /

00:40 Fitting... -

00:41 Fitting... \

00:41 Fitting... |

00:41 Fitting... /

00:41 Fitting... -

00:41 Fitting... \

00:42 Fitting... |

00:42 Fitting... /

00:42 Fitting... -

00:42 Fitting... \

00:42 Fitting... |

00:43 Fitting... /

00:43 Fitting... -

00:43 Fitting... \

00:43 Fitting... |

00:43 Fitting... /

00:44 Fitting... -

00:44 Fitting... \

00:44 Fitting... |

00:44 Fitting... /

00:44 Fitting... -

00:45 Fitting... \

00:45 Fitting... |

00:45 Fitting... /

00:45 Fitting... -

00:45 Fitting... \

00:46 Fitting... |

00:46 Fitting... /

00:46 Fitting... -

00:46 Fitting... \

00:46 Fitting... |

00:47 Fitting... /

00:47 Fitting... -

00:47 Fitting... \

00:47 Fitting... |

00:47 Fitting... /

00:48 Fitting... -

00:48 Fitting... \

00:48 Fitting... |

00:48 Fitting... /

00:48 Fitting... -

00:49 Fitting... \

00:49 Fitting... |

00:49 Fitting... /

00:49 Fitting... -

00:49 Fitting... \

00:50 Fitting... |

00:50 Fitting... /

00:50 Fitting... -

00:50 Fitting... \

00:50 Fitting... |

00:51 Fitting... /

00:51 Fitting... -

00:51 Fitting... \

00:51 Fitting... |

00:51 Fitting... /

00:52 Fitting... -

00:52 Fitting... \

00:52 Fitting... |

00:52 Fitting... /

00:52 Fitting... -

00:53 Fitting... \

00:53 Fitting... |

00:53 Fitting... /

00:53 Fitting... -

00:53 Fitting... \

00:54 Fitting... |

00:54 Fitting... /

00:54 Fitting... -

00:54 Fitting... \

00:54 Fitting... |

00:55 Fitting... /

00:55 Fitting... -

00:55 Fitting... \

00:55 Fitting... |

00:55 Fitting... /

00:56 Fitting... -

00:56 Fitting... \

00:56 Fitting... |

00:56 Fitting... /

00:56 Fitting... -

00:57 Fitting... \

00:57 Fitting... |

00:57 Fitting... /

00:57 Fitting... -

00:57 Fitting... \

00:58 Fitting... |

00:58 Fitting... /

00:58 Fitting... -

00:58 Fitting... \

00:58 Fitting... |

00:59 Fitting... /

00:59 Fitting... -

00:59 Fitting... \

00:59 Fitting... |

00:59 Fitting... /

01:00 Fitting... -

01:00 Fitting... \

01:00 Fitting... |

01:00 Fitting... /

01:00 Fitting... -

01:01 Fitting... \

01:01 Fitting... |

01:01 Fitting... /

01:01 Fitting... -

01:01 Fitting... \

01:02 Fitting... |

01:02 Fitting... /

01:02 Fitting... -

01:02 Fitting... \

01:02 Fitting... |

01:03 Fitting... /

01:03 Fitting... -

01:03 Fitting... \

01:03 Fitting... |

01:03 Fitting... /

01:04 Fitting... -

01:04 Fitting... \

01:04 Fitting... |

01:04 Fitting... /

01:04 Fitting... -

01:05 Fitting... \

01:05 Fitting... |

01:05 Fitting... /

01:05 Fitting... -

01:05 Fitting... \

01:06 Fitting... |

01:06 Fitting... /

01:06 Fitting... -

01:06 Fitting... \

01:06 Fitting... |

01:07 Fitting... /

01:07 Fitting... -

01:07 Fitting... \

01:07 Fitting... |

01:07 Fitting... /

01:08 Fitting... -

01:08 Fitting... \

01:08 Fitting... |

01:08 Fitting... /

01:08 Fitting... -

01:09 Fitting... \

01:09 Fitting... |

01:09 Fitting... /

01:09 Fitting... -

01:09 Fitting... \

01:10 Fitting... |

01:10 Fitting... /

01:10 Fitting... -

01:10 Fitting... \

01:10 Fitting... |

01:11 Fitting... /

01:11 Fitting... -

01:11 Fitting... \

01:11 Fitting... |

01:11 Fitting... /

01:12 Fitting... -

01:12 Fitting... \

01:12 Fitting... |

01:12 Fitting... /

01:12 Fitting... -

01:13 Fitting... \

01:13 Fitting... |

01:13 Fitting... /

01:13 Fitting... -

01:13 Fitting... \

01:14 Fitting... |

01:14 Fitting... /

01:14 Fitting... -

01:14 Fitting... \

01:14 Fitting... |

01:15 Fitting... /

01:15 Fitting... -

01:15 Fitting... \

01:15 Fitting... |

01:15 Fitting... /

01:16 Fitting... -

01:16 Fitting... \

01:16 Fitting... |

01:16 Fitting... /

01:16 Fitting... -

01:17 Fitting... \

01:17 Fitting... |

01:17 Fitting... /

01:17 Fitting... -

01:17 Fitting... \

01:18 Fitting... |

01:18 Fitting... /

01:18 Fitting... -

01:18 Fitting... \

01:18 Fitting... |

01:19 Fitting... /

01:19 Fitting... -

01:19 Fitting... \

01:19 Fitting... |

01:19 Fitting... /

01:20 Fitting... -

01:20 Fitting... \

01:20 Fitting... |

01:20 Fitting... /

01:20 Fitting... -

01:21 Fitting... \

01:21 Fitting... |

01:21 Fitting... /

01:21 Fitting... -

01:21 Fitting... \

01:22 Fitting... |

01:22 Fitting... /

01:22 Fitting... -

01:22 Fitting... \

01:22 Fitting... |

01:23 Fitting... /

01:23 Fitting... -

01:23 Fitting... \

01:23 Fitting... |

01:23 Fitting... /

01:24 Fitting... -

01:24 Fitting... \

01:24 Fitting... |

01:24 Fitting... /

01:24 Fitting... -

01:25 Fitting... \

01:25 Fitting... |

01:25 Fitting... /

01:25 Fitting... -

01:25 Fitting... \

01:26 Fitting... |

01:26 Fitting... /

01:26 Fitting... -

01:26 Fitting... \

01:26 Fitting... |

01:27 Fitting... /

01:27 Fitting... -

01:27 Fitting... \

01:27 Fitting... |

01:27 Fitting... /

01:28 Fitting... -

01:28 Fitting... \

01:28 Fitting... |

01:28 Fitting... /

01:28 Fitting... -

01:29 Fitting... \

01:29 Fitting... |

01:29 Fitting... /

01:29 Fitting... -

01:29 Fitting... \

01:30 Fitting... |

01:30 Fitting... /

01:30 Fitting... -

01:30 Fitting... \

01:30 Fitting... |

01:31 Fitting... /

01:31 Fitting... -

01:31 Fitting... \

01:31 Fitting... |

01:31 Fitting... /

01:32 Fitting... -

01:32 Fitting... \

01:32 Fitting... |

01:32 Fitting... /

01:32 Fitting... -

01:33 Fitting... \

01:33 Fitting... |

01:33 Fitting... /

01:33 Fitting... -

01:33 Fitting... \

01:34 Fitting... |

01:34 Fitting... /

01:34 Fitting... -

01:34 Fitting... \

01:34 Fitting... |

01:35 Fitting... /

01:35 Fitting... -

01:35 Fitting... \

01:35 Fitting... |

01:35 Fitting... /

01:36 Fitting... -

01:36 Fitting... \

01:36 Fitting... |

01:36 Fitting... /

01:36 Fitting... -

01:37 Fitting... \

01:37 Fitting... |

01:37 Fitting... /

01:37 Fitting... -

01:37 Fitting... \

01:38 Fitting... |

01:38 Fitting... /

01:38 Fitting... -

01:38 Fitting... \

01:38 Fitting... |

01:39 Fitting... /

01:39 Fitting... -

01:39 Fitting... \

01:39 Fitting... |

01:39 Fitting... /

01:40 Fitting... -

01:40 Fitting... \

01:40 Fitting... |

01:40 Fitting... /

01:40 Fitting... -

01:41 Fitting... \

01:41 Fitting... |

01:41 Fitting... /

01:41 Fitting... -

01:41 Fitting... \

01:42 Fitting... |

01:42 Fitting... /

01:42 Fitting... -

01:42 Fitting... \

01:42 Fitting... |

01:43 Fitting... /

01:43 Fitting... -

01:43 Fitting... \

01:43 Fitting... |

01:43 Fitting... /

01:44 Fitting... -

01:44 Fitting... \

01:44 Fitting... |

01:44 Fitting... /

01:44 Fitting... -

01:45 Fitting... \

01:45 Fitting... |

01:45 Fitting... /

01:45 Fitting... -

01:45 Fitting... \

01:46 Fitting... |

01:46 Fitting... /

01:46 Fitting... -

01:46 Fitting... \

01:46 Fitting... |

01:47 Fitting... /

01:47 Fitting... -

01:47 Fitting... \

01:47 Fitting... |

01:47 Fitting... /

01:48 Fitting... -

01:48 Fitting... \

01:48 Fitting... |

01:48 Fitting... /

01:48 Fitting... -

01:49 Fitting... \

01:49 Fitting... |

01:49 Fitting... /

01:49 Fitting... -

01:49 Fitting... \

01:50 Fitting... |

01:50 Fitting... /

01:50 Fitting... -

01:50 Fitting... \

01:50 Fitting... |

01:51 Fitting... /

01:51 Fitting... -

01:51 Fitting... \

01:51 Fitting... |

01:51 Fitting... /

01:52 Fitting... -

01:52 Fitting... \

01:52 Fitting... |

01:52 Fitting... /

01:52 Fitting... -

01:53 Fitting... \

01:53 Fitting... |

01:53 Fitting... /

01:53 Fitting... -

01:53 Fitting... \

01:54 Fitting... |

01:54 Fitting... /

01:54 Fitting... -

01:54 Fitting... \

01:54 Fitting... |

01:55 Fitting... /

01:55 Fitting... -

01:55 Fitting... \

01:55 Fitting... |

01:55 Fitting... /

01:56 Fitting... -

01:56 Fitting... \

01:56 Fitting... |

01:56 Fitting... /

01:56 Fitting... -

01:57 Fitting... \

01:57 Fitting... |

01:57 Fitting... /

01:57 Fitting... -

01:57 Fitting... \

01:58 Fitting... |

01:58 Fitting... /

01:58 Fitting... -

01:58 Fitting... \

01:58 Fitting... |

01:59 Fitting... /

01:59 Fitting... -

01:59 Fitting... \

01:59 Fitting... |

01:59 Fitting... /

02:00 Fitting... -

02:00 Fitting... \

02:00 Fitting... |

02:00 Fitting... /

02:00 Fitting... -

02:01 Fitting... \

02:01 Fitting... |

02:01 Fitting... /

02:01 Fitting... -

02:01 Fitting... \

02:02 Fitting... |

02:02 Fitting... /

02:02 Fitting... -

02:02 Fitting... \

02:02 Fitting... |

02:03 Fitting... /

02:03 Fitting... -

02:03 Fitting... \

02:03 Fitting... |

02:03 Fitting... /

02:04 Fitting... -

02:04 Fitting... \

02:04 Fitting... |

02:04 Fitting... /

02:04 Fitting... -

02:05 Fitting... \

02:05 Fitting... |

02:05 Fitting... /

02:05 Fitting... -

02:05 Fitting... \

02:06 Fitting... |

02:06 Fitting... /

02:06 Fitting... -

02:06 Fitting... \

02:06 Fitting... |

02:07 Fitting... /

02:07 Fitting... -

02:07 Fitting... \

02:07 Fitting... |

02:07 Fitting... /

02:08 Fitting... -

02:08 Fitting... \

02:08 Fitting... |

02:08 Fitting... /

02:08 Fitting... -

02:09 Fitting... \

02:09 Fitting... |

02:09 Fitting... /

02:09 Fitting... -

02:09 Fitting... \

02:10 Fitting... |

02:10 Fitting... /

02:10 Fitting... -

02:10 Fitting... \

02:10 Fitting... |

02:11 Fitting... /

02:11 Fitting... -

02:11 Fitting... \

02:11 Fitting... |

02:11 Fitting... /

02:12 Fitting... -

02:12 Fitting... \

02:12 Fitting... |

02:12 Fitting... /

02:12 Fitting... -

02:13 Fitting... \

02:13 Fitting... |

02:13 Fitting... /

02:13 Fitting... -

02:13 Fitting... \

02:14 Fitting... |

02:14 Fitting... /

02:14 Fitting... -

02:14 Fitting... \

02:14 Fitting... |

02:15 Fitting... /

02:15 Fitting... -

02:15 Fitting... \

02:15 Fitting... |

02:15 Fitting... /

02:16 Fitting... -

02:16 Fitting... \

02:16 Fitting... |

02:16 Fitting... /

02:16 Fitting... -

02:17 Fitting... \

02:17 Fitting... |

02:17 Fitting... /

02:17 Fitting... -

02:17 Fitting... \

02:18 Fitting... |

02:18 Fitting... /

02:18 Fitting... -

02:18 Fitting... \

02:18 Fitting... |

02:19 Fitting... /

02:19 Fitting... -

02:19 Fitting... \

02:19 Fitting... |

02:19 Fitting... /

02:20 Fitting... -

02:20 Fitting... \

02:20 Fitting... |

02:20 Fitting... /

02:20 Fitting... -

02:21 Fitting... \

02:21 Fitting... |

02:21 Fitting... /

02:21 Fitting... -

02:21 Fitting... \

02:22 Fitting... |

02:22 Fitting... /

02:22 Fitting... -

02:22 Fitting... \

02:22 Fitting... |

02:23 Fitting... /

02:23 Fitting... -

02:23 Fitting... \

02:23 Fitting... |

02:23 Fitting... /

02:24 Fitting... -

02:24 Fitting... \

02:24 Fitting... |

02:24 Fitting... /

02:24 Fitting... -

02:25 Fitting... \

02:25 Fitting... |

02:25 Fitting... /

02:25 Fitting... -

02:25 Fitting... \

02:26 Fitting... |

02:26 Fitting... /

02:26 Fitting... -

02:26 Fitting... \

02:26 Fitting... |

02:27 Fitting... /

02:27 Fitting... -

02:27 Fitting... \

02:27 Fitting... |

02:27 Fitting... /

02:28 Fitting... -

02:28 Fitting... \

02:28 Fitting... |

02:28 Fitting... /

02:28 Fitting... -

02:29 Fitting... \

02:29 Fitting... |

02:29 Fitting... /

02:29 Fitting... -

02:29 Fitting... \

02:30 Fitting... |

02:30 Fitting... /

02:30 Fitting... -

02:30 Fitting... \

02:30 Fitting... |

02:31 Fitting... /

02:31 Fitting... -

02:31 Fitting... \

02:31 Fitting... |

02:31 Fitting... /

02:32 Fitting... -

02:32 Fitting... \

02:32 Fitting... |

02:32 Fitting... /

02:32 Fitting... -

02:33 Fitting... \

02:33 Fitting... |

02:33 Fitting... /

02:33 Fitting... -

02:33 Fitting... \

02:34 Fitting... |

02:34 Fitting... /

02:34 Fitting... -

02:34 Fitting... \

02:34 Fitting... |

02:35 Fitting... /

02:35 Fitting... -

02:35 Fitting... \

02:35 Fitting... |

02:35 Fitting... /

02:36 Fitting... -

02:36 Fitting... \

02:36 Fitting... |

02:36 Fitting... /

02:36 Fitting... -

02:37 Fitting... \

02:37 Fitting... |

02:37 Fitting... /

02:37 Fitting... -

02:37 Fitting... \

02:38 Fitting... |

02:38 Fitting... /

02:38 Fitting... -

02:38 Fitting... \

02:38 Fitting... |

02:39 Fitting... /

02:39 Fitting... -

02:39 Fitting... \

02:39 Fitting... |

02:39 Fitting... /

02:40 Fitting... -

02:40 Fitting... \

02:40 Fitting... |

02:40 Fitting... /

02:40 Fitting... -

02:41 Fitting... \

02:41 Fitting... |

02:41 Fitting... /

02:41 Fitting... -

02:41 Fitting... \

02:42 Fitting... |

02:42 Fitting... /

02:42 Fitting... -

02:42 Fitting... \

02:42 Fitting... |

02:43 Fitting... /

02:43 Fitting... -

02:43 Fitting... \

02:43 Fitting... |

02:43 Fitting... /

02:44 Fitting... -

02:44 Fitting... \

02:44 Fitting... |

02:44 Fitting... /

02:44 Fitting... -

02:45 Fitting... \

02:45 Fitting... |

02:45 Fitting... /

02:45 Fitting... -

02:45 Fitting... \

02:46 Fitting... |

02:46 Fitting... /

02:46 Fitting... -

02:46 Fitting... \

02:46 Fitting... |

02:47 Fitting... /

02:47 Fitting... -

02:47 Fitting... \

02:47 Fitting... |

02:48 Fitting... /

02:48 Fitting... -

02:48 Fitting... \

02:48 Fitting... |

02:48 Fitting... /

02:49 Fitting... -

02:49 Fitting... \

02:49 Fitting... |

02:49 Fitting... /

02:49 Fitting... -

02:50 Fitting... \

02:50 Fitting... |

02:50 Fitting... /

02:50 Fitting... -

02:50 Fitting... \

02:51 Fitting... |

02:51 Fitting... /

02:51 Fitting... -

02:51 Fitting... \

02:51 Fitting... |

02:52 Fitting... /

02:52 Fitting... -

02:52 Fitting... \

02:52 Fitting... |

02:52 Fitting... /

02:53 Fitting... -

02:53 Fitting... \

02:53 Fitting... |

02:53 Fitting... /

02:53 Fitting... -

02:54 Fitting... \

02:54 Fitting... |

02:54 Fitting... /

02:54 Fitting... -

02:54 Fitting... \

02:55 Fitting... |

02:55 Fitting... /

02:55 Fitting... -

02:55 Fitting... \

02:55 Fitting... |

02:56 Fitting... /

02:56 Fitting... -

02:56 Fitting... \

02:56 Fitting... |

02:56 Fitting... /

02:57 Fitting... -

02:57 Fitting... \

02:57 Fitting... |

02:57 Fitting... /

02:57 Fitting... -

02:58 Fitting... \

02:58 Fitting... |

02:58 Fitting... /

02:58 Fitting... -

02:58 Fitting... \

02:59 Fitting... |

02:59 Fitting... /

02:59 Fitting... -

02:59 Fitting... \

02:59 Fitting... |

03:00 Fitting... /

03:00 Fitting... -

03:00 Fitting... \

03:00 Fitting... |

03:00 Fitting... /

03:01 Fitting... -

03:01 Fitting... \

03:01 Fitting... |

03:01 Fitting... /

03:01 Fitting... -

03:02 Fitting... \

03:02 Fitting... |

03:02 Fitting... /

03:02 Fitting... -

03:02 Fitting... \

03:03 Fitting... |

03:03 Fitting... /

03:03 Fitting... -

03:03 Fitting... \

03:03 Fitting... |

03:04 Fitting... /

03:04 Fitting... -

03:04 Fitting... \

03:04 Fitting... |

03:04 Fitting... /

03:05 Fitting... -

03:05 Fitting... \

03:05 Fitting... |

03:05 Fitting... /

03:05 Fitting... -

03:06 Fitting... \

03:06 Fitting... |

03:06 Fitting... /

03:06 Fitting... -

03:06 Fitting... \

03:07 Fitting... |

03:07 Fitting... /

03:07 Fitting... -

03:07 Fitting... \

03:07 Fitting... |

03:08 Fitting... /

03:08 Fitting... -

03:08 Fitting... \

03:08 Fitting... |

03:08 Fitting... /

03:09 Fitting... -

03:09 Fitting... \

03:09 Fitting... |

03:09 Fitting... /

03:09 Fitting... -

03:10 Fitting... \

03:10 Fitting... |

03:10 Fitting... /

03:10 Fitting... -

03:10 Fitting... \

03:11 Fitting... |

03:11 Fitting... /

03:11 Fitting... -

03:11 Fitting... \

03:11 Fitting... |

03:12 Fitting... /

03:12 Fitting... -

03:12 Fitting... \

03:12 Fitting... |

03:12 Fitting... /

03:13 Fitting... -

03:13 Fitting... \

03:13 Fitting... |

03:13 Fitting... /

03:13 Fitting... -

03:14 Fitting... \

03:14 Fitting... |

03:14 Fitting... /

03:14 Fitting... -

03:14 Fitting... \

03:15 Fitting... |

03:15 Fitting... /

03:15 Fitting... -

03:15 Fitting... \

03:15 Fitting... |

03:16 Fitting... /

03:16 Fitting... -

03:16 Fitting... \

03:16 Fitting... |

03:16 Fitting... /

03:17 Fitting... -

03:17 Fitting... \

03:17 Fitting... |

03:17 Fitting... /

03:17 Fitting... -

03:18 Fitting... \

03:18 Fitting... |

03:18 Fitting... /

03:18 Fitting... -

03:18 Fitting... \

03:19 Fitting... |

03:19 Fitting... /

03:19 Fitting... -

03:19 Fitting... \

03:19 Fitting... |

03:20 Fitting... /

03:20 Fitting... -

03:20 Fitting... \

03:20 Fitting... |

03:20 Fitting... /

03:21 Fitting... -

03:21 Fitting... \

03:21 Fitting... |

03:21 Fitting... /

03:21 Fitting... -

03:22 Fitting... \

03:22 Fitting... |

03:22 Fitting... /

03:22 Fitting... -

03:22 Fitting... \

03:23 Fitting... |

03:23 Fitting... /

03:23 Fitting... -

03:23 Fitting... \

03:23 Fitting... |

03:24 Fitting... /

03:24 Fitting... -

03:24 Fitting... \

03:24 Fitting... |

03:24 Fitting... /

03:25 Fitting... -

03:25 Fitting... \

03:25 Fitting... |

03:25 Fitting... /

03:25 Fitting... -

03:26 Fitting... \

03:26 Fitting... |

03:26 Fitting... /

03:26 Fitting... -

03:26 Fitting... \

03:27 Fitting... |

03:27 Fitting... /

03:27 Fitting... -

03:27 Fitting... \

03:27 Fitting... |

03:28 Fitting... /

03:28 Fitting... -

03:28 Fitting... \

03:28 Fitting... |

03:28 Fitting... /

03:29 Fitting... -

03:29 Fitting... \

03:29 Fitting... |

03:29 Fitting... /

03:29 Fitting... -

03:30 Fitting... \

03:30 Fitting... |

03:30 Fitting... /

03:30 Fitting... -

03:30 Fitting... \

03:31 Fitting... |

03:31 Fitting... /

03:31 Fitting... -

03:31 Fitting... \

03:31 Fitting... |

03:32 Fitting... /

03:32 Fitting... -

03:32 Fitting... \

03:32 Fitting... |

03:32 Fitting... /

03:33 Fitting... -

03:33 Fitting... \

03:33 Fitting... |

03:33 Fitting... /

03:33 Fitting... -

03:34 Fitting... \

03:34 Fitting... |

03:34 Fitting... /

03:34 Fitting... -

03:34 Fitting... \

03:35 Fitting... |

03:35 Fitting... /

03:35 Fitting... -

03:35 Fitting... \

03:35 Fitting... |

03:36 Fitting... /

03:36 Fitting... -

03:36 Fitting... \

03:36 Fitting... |

03:36 Fitting... /

03:37 Fitting... -

03:37 Fitting... \

03:37 Fitting... |

03:37 Fitting... /

03:37 Fitting... -

03:38 Fitting... \

03:38 Fitting... |

03:38 Fitting... /

03:38 Fitting... -

03:38 Fitting... \

03:39 Fitting... |

03:39 Fitting... /

03:39 Fitting... -

03:39 Fitting... \

03:39 Fitting... |

03:40 Fitting... /

03:40 Fitting... -

03:40 Fitting... \

03:40 Fitting... |

03:40 Fitting... /

03:41 Fitting... -

03:41 Fitting... \

03:41 Fitting... |

03:41 Fitting... /

03:41 Fitting... -

03:42 Fitting... \

03:42 Fitting... |

03:42 Fitting... /

03:42 Fitting... -

03:42 Fitting... \

03:43 Fitting... |

03:43 Fitting... /

03:43 Fitting... -

03:43 Fitting... \

03:43 Fitting... |

03:44 Fitting... /

03:44 Fitting... -

03:44 Fitting... \

03:44 Fitting... |

03:44 Fitting... /

03:45 Fitting... -

03:45 Fitting... \

03:45 Fitting... |

03:45 Fitting... /

03:45 Fitting... -

03:46 Fitting... \

03:46 Fitting... |

03:46 Fitting... /

03:46 Fitting... -

03:46 Fitting... \

03:47 Fitting... |

03:47 Fitting... /

03:47 Fitting... -

03:47 Fitting... \

03:47 Fitting... |

03:48 Fitting... /

03:48 Fitting... -

03:48 Fitting... \

03:48 Fitting... |

03:48 Fitting... /

03:49 Fitting... -

03:49 Fitting... \

03:49 Fitting... |

03:49 Fitting... /

03:49 Fitting... -

03:50 Fitting... \

03:50 Fitting... |

03:50 Fitting... /

03:50 Fitting... -

03:50 Fitting... \

03:51 Fitting... |

03:51 Fitting... /

03:51 Fitting... -

03:51 Fitting... \

03:51 Fitting... |

03:52 Fitting... /

03:52 Fitting... -

03:52 Fitting... \

03:52 Fitting... |

03:52 Fitting... /

03:53 Fitting... -

03:53 Fitting... \

03:53 Fitting... |

03:53 Fitting... /

03:53 Fitting... -

03:54 Fitting... \

03:54 Fitting... |

03:54 Fitting... /

03:54 Fitting... -

03:54 Fitting... \

03:55 Fitting... |

03:55 Fitting... /

03:55 Fitting... -

03:55 Fitting... \

03:55 Fitting... |

03:56 Fitting... /

03:56 Fitting... -

03:56 Fitting... \

03:56 Fitting... |

03:56 Fitting... /

03:57 Fitting... -

03:57 Fitting... \

03:57 Fitting... |

03:57 Fitting... /

03:57 Fitting... -

03:58 Fitting... \

03:58 Fitting... |

03:58 Fitting... /

03:58 Fitting... -

03:58 Fitting... \

03:59 Fitting... |

03:59 Fitting... /

03:59 Fitting... -

03:59 Fitting... \

03:59 Fitting... |

04:00 Fitting... /

04:00 Fitting... -

04:00 Fitting... \

04:00 Fitting... |

04:00 Fitting... /

04:01 Fitting... -

04:01 Fitting... \

04:01 Fitting... |

04:01 Fitting... /

04:01 Fitting... -

04:02 Fitting... \

04:02 Fitting... |

04:02 Fitting... /

04:02 Fitting... -

04:02 Fitting... \

04:03 Fitting... |

04:03 Fitting... /

04:03 Fitting... -

04:03 Fitting... \

04:03 Fitting... |

04:04 Fitting... /

04:04 Fitting... -

04:04 Fitting... \

04:04 Fitting... |

04:04 Fitting... /

04:05 Fitting... -

04:05 Fitting... \

04:05 Fitting... |

04:05 Fitting... /

04:05 Fitting... -

04:06 Fitting... \

04:06 Fitting... |

04:06 Fitting... /

04:06 Fitting... -

04:06 Fitting... \

04:07 Fitting... |

04:07 Fitting... /

04:07 Fitting... -

04:07 Fitting... \

04:07 Fitting... |

04:08 Fitting... /

04:08 Fitting... -

04:08 Fitting... \

04:08 Fitting... |

04:08 Fitting... /

04:09 Fitting... -

04:09 Fitting... \

04:09 Fitting... |

04:09 Fitting... /

04:09 Fitting... -

04:10 Fitting... \

04:10 Fitting... |

04:10 Fitting... /

04:10 Fitting... -

04:10 Fitting... \

04:11 Fitting... |

04:11 Fitting... /

04:11 Fitting... -

04:11 Fitting... \

04:11 Fitting... |

04:12 Fitting... /

04:12 Fitting... -

04:12 Fitting... \

04:12 Fitting... |

04:12 Fitting... /

04:13 Fitting... -

04:13 Fitting... \

04:13 Fitting... |

04:13 Fitting... /

04:13 Fitting... -

04:14 Fitting... \

04:14 Fitting... |

04:14 Fitting... /

04:14 Fitting... -

04:14 Fitting... \

04:15 Fitting... |

04:15 Fitting... /

04:15 Fitting... -

04:15 Fitting... \

04:15 Fitting... |

04:16 Fitting... /

04:16 Fitting... -

04:16 Fitting... \

04:16 Fitting... |

04:16 Fitting... /

04:17 Fitting... -

04:17 Fitting... \

04:17 Fitting... |

04:17 Fitting... /

04:17 Fitting... -

04:18 Fitting... \

04:18 Fitting... |

04:18 Fitting... /

04:18 Fitting... -

04:18 Fitting... \

04:19 Fitting... |

04:19 Fitting... /

04:19 Fitting... -

04:19 Fitting... \

04:19 Fitting... |

04:20 Fitting... /

04:20 Fitting... -

04:20 Fitting... \

04:20 Fitting... |

04:20 Fitting... /

04:21 Fitting... -

04:21 Fitting... \

04:21 Fitting... |

04:21 Fitting... /

04:21 Fitting... -

04:22 Fitting... \

04:22 Fitting... |

04:22 Fitting... /

04:22 Fitting... -

04:22 Fitting... \

04:23 Fitting... |

04:23 Fitting... /

04:23 Fitting... -

04:23 Fitting... \

04:23 Fitting... |

04:24 Fitting... /

04:24 Fitting... -

04:24 Fitting... \

04:24 Fitting... |

04:24 Fitting... /

04:25 Fitting... -

04:25 Fitting... \

04:25 Fitting... |

04:25 Fitting... /

04:25 Fitting... -

04:26 Fitting... \

04:26 Fitting... |

04:26 Fitting... /

04:26 Fitting... -

04:26 Fitting... \

04:27 Fitting... |

04:27 Fitting... /

04:27 Fitting... -

04:27 Fitting... \

04:27 Fitting... |

04:28 Fitting... /

04:28 Fitting... -

04:28 Fitting... \

04:28 Fitting... |

04:28 Fitting... /

04:29 Fitting... -

04:29 Fitting... \

04:29 Fitting... |

04:29 Fitting... /

04:29 Fitting... -

04:30 Fitting... \

04:30 Fitting... |

04:30 Fitting... /

04:30 Fitting... -

04:30 Fitting... \

04:31 Fitting... |

04:31 Fitting... /

04:31 Fitting... -

04:31 Fitting... \

04:31 Fitting... |

04:32 Fitting... /

04:32 Fitting... -

04:32 Fitting... \

04:32 Fitting... |

04:32 Fitting... /

04:33 Fitting... -

04:33 Fitting... \

04:33 Fitting... |

04:33 Fitting... /

04:33 Fitting... -

04:34 Fitting... \

04:34 Fitting... |

04:34 Fitting... /

04:34 Fitting... -

04:34 Fitting... \

04:35 Fitting... |

04:35 Fitting... /

04:35 Fitting... -

04:35 Fitting... \

04:35 Fitting... |

04:36 Fitting... /

04:36 Fitting... -

04:36 Fitting... \

04:36 Fitting... |

04:36 Fitting... /

04:37 Fitting... -

04:37 Fitting... \

04:37 Fitting... |

04:37 Fitting... /

04:37 Fitting... -

04:38 Fitting... \

04:38 Fitting... |

04:38 Fitting... /

04:38 Fitting... -

04:38 Fitting... \

04:39 Fitting... |

04:39 Fitting... /

04:39 Fitting... -

04:39 Fitting... \

04:39 Fitting... |

04:40 Fitting... /

04:40 Fitting... -

04:40 Fitting... \

04:40 Fitting... |

04:40 Fitting... /

04:41 Fitting... -

04:41 Fitting... \

04:41 Fitting... |

04:41 Fitting... /

04:41 Fitting... -

04:42 Fitting... \

04:42 Fitting... |

04:42 Fitting... /

04:42 Fitting... -

04:42 Fitting... \

04:43 Fitting... |

04:43 Fitting... /

04:43 Fitting... -

04:43 Fitting... \

04:43 Fitting... |

04:44 Fitting... /

04:44 Fitting... -

04:44 Fitting... \

04:44 Fitting... |

04:44 Fitting... /

04:45 Fitting... -

04:45 Fitting... \

04:45 Fitting... |

04:45 Fitting... /

04:45 Fitting... -

04:46 Fitting... \

04:46 Fitting... |

04:46 Fitting... /

04:46 Fitting... -

04:46 Fitting... \

04:47 Fitting... |

04:47 Fitting... /

04:47 Fitting... -

04:47 Fitting... \

04:47 Fitting... |

04:48 Fitting... /

04:48 Fitting... -

04:48 Fitting... \

04:48 Fitting... |

04:48 Fitting... /

04:49 Fitting... -

04:49 Fitting... \

04:49 Fitting... |

04:49 Fitting... /

04:49 Fitting... -

04:50 Fitting... \

04:50 Fitting... |

04:50 Fitting... /

04:50 Fitting... -

04:50 Fitting... \

04:51 Fitting... |

04:51 Fitting... /

04:51 Fitting... -

04:51 Fitting... \

04:51 Fitting... |

04:52 Fitting... /

04:52 Fitting... -

04:52 Fitting... \

04:52 Fitting... |

04:52 Fitting... /

04:53 Fitting... -

04:53 Fitting... \

04:53 Fitting... |

04:53 Fitting... /

04:53 Fitting... -

04:54 Fitting... \

04:54 Fitting... |

04:54 Fitting... /

04:54 Fitting... -

04:54 Fitting... \

04:55 Fitting... |

04:55 Fitting... /

04:55 Fitting... -

04:55 Fitting... \

04:55 Fitting... |

04:56 Fitting... /

04:56 Fitting... -

04:56 Fitting... \

04:56 Fitting... |

04:56 Fitting... /

04:57 Fitting... -

04:57 Fitting... \

04:57 Fitting... |

04:57 Fitting... /

04:57 Fitting... -

04:58 Fitting... \

04:58 Fitting... |

04:58 Fitting... /

04:58 Fitting... -

04:58 Fitting... \

04:59 Fitting... |

04:59 Fitting... /

04:59 Fitting... -

04:59 Fitting... \

04:59 Fitting... |

05:00 Fitting... /

05:00 Fitting... -

05:00 Fitting... \

05:00 Fitting... |

05:00 Fitting... /

05:01 Fitting... -

05:01 Fitting... \

05:01 Fitting... |

05:01 Fitting... /

05:01 Fitting... -

05:02 Fitting... \

05:02 Fitting... |

05:02 Fitting... /

05:02 Fitting... -

05:02 Fitting... \

05:03 Fitting... |

05:03 Fitting... /

05:03 Fitting... -

05:03 Fitting... \

05:03 Fitting... |

05:04 Fitting... /

05:04 Fitting... -

05:04 Fitting... \

05:04 Fitting... |

05:04 Fitting... /

05:05 Fitting... -

05:05 Fitting... \

05:05 Fitting... |

05:05 Fitting... /

05:05 Fitting... -

05:06 Fitting... \

05:06 Fitting... |

05:06 Fitting... /

05:06 Fitting... -

05:06 Fitting... \

05:07 Fitting... |

05:07 Fitting... /

05:07 Fitting... -

05:07 Fitting... \

05:07 Fitting... |

05:08 Fitting... /

05:08 Fitting... -

05:08 Fitting... \

05:08 Fitting... |

05:08 Fitting... /

05:09 Fitting... -

05:09 Fitting... \

05:09 Fitting... |

05:09 Fitting... /

05:09 Fitting... -

05:10 Fitting... \

05:10 Fitting... |

05:10 Fitting... /

05:10 Fitting... -

05:10 Fitting... \

05:11 Fitting... |

05:11 Fitting... /

05:11 Fitting... -

05:11 Fitting... \

05:11 Fitting... |

05:12 Fitting... /

05:12 Fitting... -

05:12 Fitting... \

05:12 Fitting... |

05:12 Fitting... /

05:13 Fitting... -

05:13 Fitting... \

05:13 Fitting... |

05:13 Fitting... /

05:13 Fitting... -

05:14 Fitting... \

05:14 Fitting... |

05:14 Fitting... /

05:14 Fitting... -

05:14 Fitting... \

05:15 Fitting... |

05:15 Fitting... /

05:15 Fitting... -

05:15 Fitting... \

05:15 Fitting... |

05:16 Fitting... /

05:16 Fitting... -

05:16 Fitting... \

05:16 Fitting... |

05:16 Fitting... /

05:17 Fitting... -

05:17 Fitting... \

05:17 Fitting... |

05:17 Fitting... /

05:17 Fitting... -

05:18 Fitting... \

05:18 Fitting... |

05:18 Fitting... /

05:18 Fitting... -

05:18 Fitting... \

05:19 Fitting... |

05:19 Fitting... /

05:19 Fitting... -

05:19 Fitting... \

05:19 Fitting... |

05:20 Fitting... /

05:20 Fitting... -

05:20 Fitting... \

05:20 Fitting... |

05:20 Fitting... /

05:21 Fitting... -

05:21 Fitting... \

05:21 Fitting... |

05:21 Fitting... /

05:21 Fitting... -

05:22 Fitting... \

05:22 Fitting... |

05:22 Fitting... /

05:22 Fitting... -

05:22 Fitting... \

05:23 Fitting... |

05:23 Fitting... /

05:23 Fitting... -

05:23 Fitting... \

05:23 Fitting... |

05:24 Fitting... /

05:24 Fitting... -

05:24 Fitting... \

05:24 Fitting... |

05:24 Fitting... /

05:25 Fitting... -

05:25 Fitting... \

05:25 Fitting... |

05:25 Fitting... /

05:25 Fitting... -

05:26 Fitting... \

05:26 Fitting... |

05:26 Fitting... /

05:26 Fitting... -

05:26 Fitting... \

05:27 Fitting... |

05:27 Fitting... /

05:27 Fitting... -

05:27 Fitting... \

05:27 Fitting... |

05:28 Fitting... /

05:28 Fitting... -

05:28 Fitting... \

05:28 Fitting... |

05:28 Fitting... /

05:29 Fitting... -

05:29 Fitting... \

05:29 Fitting... |

05:29 Fitting... /

05:29 Fitting... -

05:30 Fitting... \

05:30 Fitting... |

05:30 Fitting... /

05:30 Fitting... -

05:30 Fitting... \

05:31 Fitting... |

05:31 Fitting... /

05:31 Fitting... -

05:31 Fitting... \

05:31 Fitting... |

05:32 Fitting... /

05:32 Fitting... -

05:32 Fitting... \

05:32 Fitting... |

05:32 Fitting... /

05:33 Fitting... -

05:33 Fitting... \

05:33 Fitting... |

05:33 Fitting... /

05:33 Fitting... -

05:34 Fitting... \

05:34 Fitting... |

05:34 Fitting... /

05:34 Fitting... -

05:34 Fitting... \

05:35 Fitting... |

05:35 Fitting... /

05:35 Fitting... -

05:35 Fitting... \

05:35 Fitting... |

05:36 Fitting... /

05:36 Fitting... -

05:36 Fitting... \

05:36 Fitting... |

05:36 Fitting... /

05:37 Fitting... -

05:37 Fitting... \

05:37 Fitting... |

05:37 Fitting... /

05:37 Fitting... -

05:38 Fitting... \

05:38 Fitting... |

05:38 Fitting... /

05:38 Fitting... -

05:38 Fitting... \

05:39 Fitting... |

05:39 Fitting... /

05:39 Fitting... -

05:39 Fitting... \

05:39 Fitting... |

05:40 Fitting... /

05:40 Fitting... -

05:40 Fitting... \

05:40 Fitting... |

05:40 Fitting... /

05:41 Fitting... -

05:41 Fitting... \

05:41 Fitting... |

05:41 Fitting... /

05:41 Fitting... -

05:42 Fitting... \

05:42 Fitting... |

05:42 Fitting... /

05:42 Fitting... -

05:42 Fitting... \

05:43 Fitting... |

05:43 Fitting... /

05:43 Fitting... -

05:43 Fitting... \

05:43 Fitting... |

05:44 Fitting... /

05:44 Fitting... -

05:44 Fitting... \

05:44 Fitting... |

05:44 Fitting... /

05:45 Fitting... -

05:45 Fitting... \

05:45 Fitting... |

05:45 Fitting... /

05:45 Fitting... -

05:46 Fitting... \

05:46 Fitting... |

05:46 Fitting... /

05:46 Fitting... -

05:46 Fitting... \

05:47 Fitting... |

05:47 Fitting... /

05:47 Fitting... -

05:47 Fitting... \

05:47 Fitting... |

05:48 Fitting... /

05:48 Fitting... -

05:48 Fitting... \

05:48 Fitting... |

05:48 Fitting... /

05:49 Fitting... -

05:49 Fitting... \

05:49 Fitting... |

05:49 Fitting... /

05:49 Fitting... -

05:50 Fitting... \

05:50 Fitting... |

05:50 Fitting... /

05:50 Fitting... -

05:50 Fitting... \

05:51 Fitting... |

05:51 Fitting... /

05:51 Fitting... -

05:51 Fitting... \

05:51 Fitting... |

05:52 Fitting... /

05:52 Fitting... -

05:52 Fitting... \

05:52 Fitting... |

05:52 Fitting... /

05:53 Fitting... -

05:53 Fitting... \

05:53 Fitting... |

05:53 Fitting... /

05:53 Fitting... -

05:54 Fitting... \

05:54 Fitting... |

05:54 Fitting... /

05:54 Fitting... -

05:54 Fitting... \

05:55 Fitting... |

05:55 Fitting... /

05:55 Fitting... -

05:55 Fitting... \

05:55 Fitting... |

05:56 Fitting... /

05:56 Fitting... -

05:56 Fitting... \

05:56 Fitting... |

05:56 Fitting... /

05:57 Fitting... -

05:57 Fitting... \

05:57 Fitting... |

05:57 Fitting... /

05:57 Fitting... -

05:58 Fitting... \

05:58 Fitting... |

05:58 Fitting... /

05:58 Fitting... -

05:58 Fitting... \

05:59 Fitting... |

05:59 Fitting... /

05:59 Fitting... -

05:59 Fitting... \

05:59 Fitting... |

06:00 Fitting... /

06:00 Fitting... -

06:00 Fitting... \

06:00 Fitting... |

06:00 Fitting... /

06:01 Fitting... -

06:01 Fitting... \

06:01 Fitting... |

06:01 Fitting... /

06:02 Fitting... -

06:02 Fitting... \

06:02 Fitting... |

06:02 Fitting... /

06:02 Fitting... -

06:03 Fitting... \

06:03 Fitting... |

06:03 Fitting... /

06:03 Fitting... -

06:03 Fitting... \

06:04 Fitting... |

06:04 Fitting... /

06:04 Fitting... -

06:04 Fitting... \

06:04 Fitting... |

06:05 Fitting... /

06:05 Fitting... -

06:05 Fitting... \

06:05 Fitting... |

06:05 Fitting... /

06:06 Fitting... -

06:06 Fitting... \

06:06 Fitting... |

06:06 Fitting... /

06:06 Fitting... -

06:07 Fitting... \

06:07 Fitting... |

06:07 Fitting... /

06:07 Fitting... -

06:07 Fitting... \

06:08 Fitting... |

06:08 Fitting... /

06:08 Fitting... -

06:08 Fitting... \

06:08 Fitting... |

06:09 Fitting... /

06:09 Fitting... -

06:09 Fitting... \

06:09 Fitting... |

06:09 Fitting... /

06:10 Fitting... -

06:10 Fitting... \

06:10 Fitting... |

06:10 Fitting... /

06:10 Fitting... -

06:11 Fitting... \

06:11 Fitting... |

06:11 Fitting... /

06:11 Fitting... -

06:11 Fitting... \

06:12 Fitting... |

06:12 Fitting... /

06:12 Fitting... -

06:12 Fitting... \

06:12 Fitting... |

06:13 Fitting... /

06:13 Fitting... -

06:13 Fitting... \

06:13 Fitting... |

06:13 Fitting... /

06:14 Fitting... -

06:14 Fitting... \

06:14 Fitting... |

06:14 Fitting... /

06:14 Fitting... -

06:15 Fitting... \

06:15 Fitting... |

06:15 Fitting... /

06:15 Fitting... -

06:15 Fitting... \

06:16 Fitting... |

06:16 Fitting... /

06:16 Fitting... -

06:16 Fitting... \

06:16 Fitting... |

06:17 Fitting... /

06:17 Fitting... -

06:17 Fitting... \

06:17 Fitting... |

06:17 Fitting... /

06:18 Fitting... -

06:18 Fitting... \

06:18 Fitting... |

06:18 Fitting... /

06:18 Fitting... -

06:19 Fitting... \

06:19 Fitting... |

06:19 Fitting... /

06:19 Fitting... -

06:19 Fitting... \

06:20 Fitting... |

06:20 Fitting... /

06:20 Fitting... -

06:20 Fitting... \

06:20 Fitting... |

06:21 Fitting... /

06:21 Fitting... -

06:21 Fitting... \

06:21 Fitting... |

06:21 Fitting... /

06:22 Fitting... -

06:22 Fitting... \

06:22 Fitting... |

06:22 Fitting... /

06:22 Fitting... -

06:23 Fitting... \

06:23 Fitting... |

06:23 Fitting... /

06:23 Fitting... -

06:23 Fitting... \

06:24 Fitting... |

06:24 Fitting... /

06:24 Fitting... -

06:24 Fitting... \

06:24 Fitting... |

06:25 Fitting... /

06:25 Fitting... -

06:25 Fitting... \

06:25 Fitting... |

06:25 Fitting... /

06:26 Fitting... -

06:26 Fitting... \

06:26 Fitting... |

06:26 Fitting... /

06:26 Fitting... -

06:27 Fitting... \

06:27 Fitting... |

06:27 Fitting... /

06:27 Fitting... -

06:27 Fitting... \

06:28 Fitting... |

06:28 Fitting... /

06:28 Fitting... -

06:28 Fitting... \

06:28 Fitting... |

06:29 Fitting... /

06:29 Fitting... -

06:29 Fitting... \

06:29 Fitting... |

06:29 Fitting... /

06:30 Fitting... -

06:30 Fitting... \

06:30 Fitting... |

06:30 Fitting... /

06:30 Fitting... -

06:31 Fitting... \

06:31 Fitting... |

06:31 Fitting... /

06:31 Fitting... -

06:31 Fitting... \

06:32 Fitting... |

06:32 Fitting... /

06:32 Fitting... -

06:32 Fitting... \

06:32 Fitting... |

06:33 Fitting... /

06:33 Fitting... -

06:33 Fitting... \

06:33 Fitting... |

06:33 Fitting... /

06:34 Fitting... -

06:34 Fitting... \

06:34 Fitting... |

06:34 Fitting... /

06:34 Fitting... -

06:35 Fitting... \

06:35 Fitting... |

06:35 Fitting... /

06:35 Fitting... -

06:35 Fitting... \

06:36 Fitting... |

06:36 Fitting... /

06:36 Fitting... -

06:36 Fitting... \

06:36 Fitting... |

06:37 Fitting... /

06:37 Fitting... -

06:37 Fitting... \

06:37 Fitting... |

06:37 Fitting... /

06:38 Fitting... -

06:38 Fitting... \

06:38 Fitting... |

06:38 Fitting... /

06:38 Fitting... -

06:39 Fitting... \

06:39 Fitting... |

06:39 Fitting... /

06:39 Fitting... -

06:39 Fitting... \

06:40 Fitting... |

06:40 Fitting... /

06:40 Fitting... -

06:40 Fitting... \

06:40 Fitting... |

06:41 Fitting... /

06:41 Fitting... -

06:41 Fitting... \

06:41 Fitting... |

06:41 Fitting... /

06:42 Fitting... -

06:42 Fitting... \

06:42 Fitting... |

06:42 Fitting... /

06:42 Fitting... -

06:43 Fitting... \

06:43 Fitting... |

06:43 Fitting... /

06:43 Fitting... -

06:43 Fitting... \

06:44 Fitting... |

06:44 Fitting... /

06:44 Fitting... -

06:44 Fitting... \

06:44 Fitting... |

06:45 Fitting... /

06:45 Fitting... -

06:45 Fitting... \

06:45 Fitting... |

06:45 Fitting... /

06:46 Fitting... -

06:46 Fitting... \

06:46 Fitting... |

06:46 Fitting... /

06:46 Fitting... -

06:47 Fitting... \

06:47 Fitting... |

06:47 Fitting... /

06:47 Fitting... -

06:47 Fitting... \

06:48 Fitting... |

06:48 Fitting... /

06:48 Fitting... -

06:48 Fitting... \

06:48 Fitting... |

06:49 Fitting... /

06:49 Fitting... -

06:49 Fitting... \

06:49 Fitting... |

06:49 Fitting... /

06:50 Fitting... -

06:50 Fitting... \

06:50 Fitting... |

06:50 Fitting... /

06:50 Fitting... -

06:51 Fitting... \

06:51 Fitting... |

06:51 Fitting... /

06:51 Fitting... -

06:51 Fitting... \

06:52 Fitting... |

06:52 Fitting... /

06:52 Fitting... -

06:52 Fitting... \

06:52 Fitting... |

06:53 Fitting... /

06:53 Fitting... -

06:53 Fitting... \

06:53 Fitting... |

06:53 Fitting... /

06:54 Fitting... -

06:54 Fitting... \

06:54 Fitting... |

06:54 Fitting... /

06:54 Fitting... -

06:55 Fitting... \

06:55 Fitting... |

06:55 Fitting... /

06:55 Fitting... -

06:55 Fitting... \

06:56 Fitting... |

06:56 Fitting... /

06:56 Fitting... -

06:56 Fitting... \

06:56 Fitting... |

06:57 Fitting... /

06:57 Fitting... -

06:57 Fitting... \

06:57 Fitting... |

06:57 Fitting... /

06:58 Fitting... -

06:58 Fitting... \

06:58 Fitting... |

06:58 Fitting... /

06:58 Fitting... -

06:59 Fitting... \

06:59 Fitting... |

06:59 Fitting... /

06:59 Fitting... -

06:59 Fitting... \

07:00 Fitting... |

07:00 Fitting... /

07:00 Fitting... -

07:00 Fitting... \

07:00 Fitting... |

07:01 Fitting... /

07:01 Fitting... -

07:01 Fitting... \

07:01 Fitting... |

07:01 Fitting... /

07:02 Fitting... -

07:02 Fitting... \

07:02 Fitting... |

07:02 Fitting... /

07:02 Fitting... -

07:03 Fitting... \

07:03 Fitting... |

07:03 Fitting... /

07:03 Fitting... -

07:03 Fitting... \

07:04 Fitting... |

07:04 Fitting... /

07:04 Fitting... -

07:04 Fitting... \

07:04 Fitting... |

07:05 Fitting... /

07:05 Fitting... -

07:05 Fitting... \

07:05 Fitting... |

07:05 Fitting... /

07:06 Fitting... -

07:06 Fitting... \

07:06 Fitting... |

07:06 Fitting... /

07:06 Fitting... -

07:07 Fitting... \

07:07 Fitting... |

07:07 Fitting... /

07:07 Fitting... -

07:07 Fitting... \

07:08 Fitting... |

07:08 Fitting... /

07:08 Fitting... -

07:08 Fitting... \

07:08 Fitting... |

07:09 Fitting... /

07:09 Fitting... -

07:09 Fitting... \

07:09 Fitting... |

07:09 Fitting... /

07:10 Fitting... -

07:10 Fitting... \

07:10 Fitting... |

07:10 Fitting... /

07:10 Fitting... -

07:11 Fitting... \

07:11 Fitting... |

07:11 Fitting... /

07:11 Fitting... -

07:11 Fitting... \

07:12 Fitting... |

07:12 Fitting... /

07:12 Fitting... -

07:12 Fitting... \

07:12 Fitting... |

07:13 Fitting... /

07:13 Fitting... -

07:13 Fitting... \

07:13 Fitting... |

07:13 Fitting... /

07:14 Fitting... -

07:14 Fitting... \

07:14 Fitting... |

07:14 Fitting... /

07:14 Fitting... -

07:15 Fitting... \

07:15 Fitting... |

07:15 Fitting... /

07:15 Fitting... -

07:15 Fitting... \

07:16 Fitting... |

07:16 Fitting... /

07:16 Fitting... -

07:16 Fitting... \

07:16 Fitting... |

07:17 Fitting... /

07:17 Fitting... -

07:17 Fitting... \

07:17 Fitting... |

07:17 Fitting... /

07:18 Fitting... -

07:18 Fitting... \

07:18 Fitting... |

07:18 Fitting... /

07:18 Fitting... -

07:19 Fitting... \

07:19 Fitting... |

07:19 Fitting... /

07:19 Fitting... -

07:19 Fitting... \

07:20 Fitting... |

07:20 Fitting... /

07:20 Fitting... -

07:20 Fitting... \

07:20 Fitting... |

07:21 Fitting... /

07:21 Fitting... -

07:21 Fitting... \

07:21 Fitting... |

07:21 Fitting... /

07:22 Fitting... -

07:22 Fitting... \

07:22 Fitting... |

07:22 Fitting... /

07:22 Fitting... -

07:23 Fitting... \

07:23 Fitting... |

07:23 Fitting... /

07:23 Fitting... -

07:23 Fitting... \

07:24 Fitting... |

07:24 Fitting... /

07:24 Fitting... -

07:24 Fitting... \

07:24 Fitting... |

07:25 Fitting... /

07:25 Fitting... -

07:25 Fitting... \

07:25 Fitting... |

07:25 Fitting... /

07:26 Fitting... -

07:26 Fitting... \

07:26 Fitting... |

07:26 Fitting... /

07:26 Fitting... -

07:27 Fitting... \

07:27 Fitting... |

07:27 Fitting... /

07:27 Fitting... -

07:27 Fitting... \

07:28 Fitting... |

07:28 Fitting... /

07:28 Fitting... -

07:28 Fitting... \

07:28 Fitting... |

07:29 Fitting... /

07:29 Fitting... -

07:29 Fitting... \

07:29 Fitting... |

07:29 Fitting... /

07:30 Fitting... -

07:30 Fitting... \

07:30 Fitting... |

07:30 Fitting... /

07:30 Fitting... -

07:31 Fitting... \

07:31 Fitting... |

07:31 Fitting... /

07:31 Fitting... -

07:31 Fitting... \

07:32 Fitting... |

07:32 Fitting... /

07:32 Fitting... -

07:32 Fitting... \

07:32 Fitting... |

07:33 Fitting... /

07:33 Fitting... -

07:33 Fitting... \

07:33 Fitting... |

07:33 Fitting... /

07:34 Fitting... -

07:34 Fitting... \

07:34 Fitting... |

07:34 Fitting... /

07:34 Fitting... -

07:35 Fitting... \

07:35 Fitting... |

07:35 Fitting... /

07:35 Fitting... -

07:35 Fitting... \

07:36 Fitting... |

07:36 Fitting... /

07:36 Fitting... -

07:36 Fitting... \

07:36 Fitting... |

07:37 Fitting... /

07:37 Fitting... -

07:37 Fitting... \

07:37 Fitting... |

07:37 Fitting... /

07:38 Fitting... -

07:38 Fitting... \

07:38 Fitting... |

07:38 Fitting... /

07:38 Fitting... -

07:39 Fitting... \

07:39 Fitting... |

07:39 Fitting... /

07:39 Fitting... -

07:39 Fitting... \

07:40 Fitting... |

07:40 Fitting... /

07:40 Fitting... -

07:40 Fitting... \

07:40 Fitting... |

07:41 Fitting... /

07:41 Fitting... -

07:41 Fitting... \

07:41 Fitting... |

07:41 Fitting... /

07:42 Fitting... -

07:42 Fitting... \

07:42 Fitting... |

07:42 Fitting... /

07:42 Fitting... -

07:43 Fitting... \

07:43 Fitting... |

07:43 Fitting... /

07:43 Fitting... -

07:43 Fitting... \

07:44 Fitting... |

07:44 Fitting... /

07:44 Fitting... -

07:44 Fitting... \

07:44 Fitting... |

07:45 Fitting... /

07:45 Fitting... -

07:45 Fitting... \

07:45 Fitting... |

07:45 Fitting... /

07:46 Fitting... -

07:46 Fitting... \

07:46 Fitting... |

07:46 Fitting... /

07:46 Fitting... -

07:47 Fitting... \

07:47 Fitting... |

07:47 Fitting... /

07:47 Fitting... -

07:47 Fitting... \

07:48 Fitting... |

07:48 Fitting... /

07:48 Fitting... -

07:48 Fitting... \

07:48 Fitting... |

07:49 Fitting... /

07:49 Fitting... -

07:49 Fitting... \

07:49 Fitting... |

07:49 Fitting... /

07:50 Fitting... -

07:50 Fitting... \

07:50 Fitting... |

07:50 Fitting... /

07:50 Fitting... -

07:51 Fitting... \

07:51 Fitting... |

07:51 Fitting... /

07:51 Fitting... -

07:51 Fitting... \

07:52 Fitting... |

07:52 Fitting... /

07:52 Fitting... -

07:52 Fitting... \

07:52 Fitting... |

07:53 Fitting... /

07:53 Fitting... -

07:53 Fitting... \

07:53 Fitting... |

07:53 Fitting... /

07:54 Fitting... -

07:54 Fitting... \

07:54 Fitting... |

07:54 Fitting... /

07:54 Fitting... -

07:55 Fitting... \

07:55 Fitting... |

07:55 Fitting... /

07:55 Fitting... -

07:55 Fitting... \

07:56 Fitting... |

07:56 Fitting... /

07:56 Fitting... -

07:56 Fitting... \

07:56 Fitting... |

07:57 Fitting... /

07:57 Fitting... -

07:57 Fitting... \

07:57 Fitting... |

07:57 Fitting... /

07:58 Fitting... -

07:58 Fitting... \

07:58 Fitting... |

07:58 Fitting... /

07:58 Fitting... -

07:59 Fitting... \

07:59 Fitting... |

07:59 Fitting... /

07:59 Fitting... -

07:59 Fitting... \

08:00 Fitting... |

08:00 Fitting... /

08:00 Fitting... -

08:00 Fitting... \

08:00 Fitting... |

08:01 Fitting... /

08:01 Fitting... -

08:01 Fitting... \

08:01 Fitting... |

08:01 Fitting... /

08:02 Fitting... -

08:02 Fitting... \

08:02 Fitting... |

08:02 Fitting... /

08:02 Fitting... -

08:03 Fitting... \

08:03 Fitting... |

08:03 Fitting... /

08:03 Fitting... -

08:03 Fitting... \

08:04 Fitting... |

08:04 Fitting... /

08:04 Fitting... -

08:04 Fitting... \

08:04 Fitting... |

08:05 Fitting... /

08:05 Fitting... -

08:05 Fitting... \

08:05 Fitting... |

08:05 Fitting... /

08:06 Fitting... -

08:06 Fitting... \

08:06 Fitting... |

08:06 Fitting... /

08:06 Fitting... -

08:06 Fitting... Done!


00:00 Predicting... -

00:00 Predicting... \

00:00 Predicting... |

00:00 Predicting... /

00:00 Predicting... -

00:01 Predicting... \

00:01 Predicting... |

00:01 Predicting... /

00:01 Predicting... -

00:01 Predicting... \

00:02 Predicting... |

00:02 Predicting... /

00:02 Predicting... -

00:02 Predicting... \

00:02 Predicting... |

00:03 Predicting... /

00:03 Predicting... -

00:03 Predicting... \

00:03 Predicting... |

00:03 Predicting... /

00:04 Predicting... -

00:04 Predicting... \

00:04 Predicting... |

00:04 Predicting... /

00:04 Predicting... -

00:05 Predicting... \

00:05 Predicting... |

00:05 Predicting... /

00:05 Predicting... -

00:05 Predicting... \

00:06 Predicting... |

00:06 Predicting... /

00:06 Predicting... -

00:06 Predicting... \

00:06 Predicting... |

00:07 Predicting... /

00:07 Predicting... -

00:07 Predicting... \

00:07 Predicting... |

00:07 Predicting... /

00:08 Predicting... -

00:08 Predicting... \

00:08 Predicting... |

00:08 Predicting... /

00:08 Predicting... -

00:09 Predicting... \

00:09 Predicting... |

00:09 Predicting... /

00:09 Predicting... -

00:09 Predicting... \

00:10 Predicting... |

00:10 Predicting... /

00:10 Predicting... -

00:10 Predicting... \

00:10 Predicting... |

00:11 Predicting... /

00:11 Predicting... -

00:11 Predicting... \

00:11 Predicting... |

00:11 Predicting... /

00:12 Predicting... -

00:12 Predicting... \

00:12 Predicting... |

00:12 Predicting... /

00:12 Predicting... -

00:13 Predicting... \

00:13 Predicting... |

00:13 Predicting... /

00:13 Predicting... -

00:13 Predicting... \

00:14 Predicting... |

00:14 Predicting... /

00:14 Predicting... -

00:14 Predicting... \

00:14 Predicting... |

00:15 Predicting... /

00:15 Predicting... -

00:15 Predicting... \

00:15 Predicting... |

00:15 Predicting... /

00:16 Predicting... -

00:16 Predicting... \

00:16 Predicting... |

00:16 Predicting... /

00:16 Predicting... -

00:17 Predicting... \

00:17 Predicting... |

00:17 Predicting... /

00:17 Predicting... -

00:17 Predicting... \

00:18 Predicting... |

00:18 Predicting... /

00:18 Predicting... -

00:18 Predicting... \

00:18 Predicting... |

00:19 Predicting... /

00:19 Predicting... -

00:19 Predicting... \

00:19 Predicting... |

00:19 Predicting... /

00:20 Predicting... -

00:20 Predicting... \

00:20 Predicting... |

00:20 Predicting... /

00:20 Predicting... -

00:21 Predicting... \

00:21 Predicting... |

00:21 Predicting... /

00:21 Predicting... -

00:21 Predicting... \

00:22 Predicting... |

00:22 Predicting... /

00:22 Predicting... -

00:22 Predicting... \

00:22 Predicting... |

00:23 Predicting... /

00:23 Predicting... -

00:23 Predicting... \

00:23 Predicting... |

00:23 Predicting... /

00:24 Predicting... -

00:24 Predicting... \

00:24 Predicting... |

00:24 Predicting... /

00:24 Predicting... -

00:25 Predicting... \

00:25 Predicting... |

00:25 Predicting... /

00:25 Predicting... -

00:25 Predicting... \

00:26 Predicting... |

00:26 Predicting... /

00:26 Predicting... -

00:26 Predicting... \

00:26 Predicting... |

00:27 Predicting... /

00:27 Predicting... -

00:27 Predicting... \

00:27 Predicting... |

00:27 Predicting... /

00:28 Predicting... -

00:28 Predicting... \

00:28 Predicting... |

00:28 Predicting... /

00:28 Predicting... -

00:29 Predicting... \

00:29 Predicting... |

00:29 Predicting... /

00:29 Predicting... -

00:29 Predicting... \

00:30 Predicting... |

00:30 Predicting... /

00:30 Predicting... -

00:30 Predicting... \

00:30 Predicting... |

00:31 Predicting... /

00:31 Predicting... -

00:31 Predicting... \

00:31 Predicting... |

00:31 Predicting... /

00:32 Predicting... -

00:32 Predicting... \

00:32 Predicting... |

00:32 Predicting... /

00:32 Predicting... -

00:33 Predicting... \

00:33 Predicting... |

00:33 Predicting... /

00:33 Predicting... -

00:33 Predicting... \

00:34 Predicting... |

00:34 Predicting... /

00:34 Predicting... -

00:34 Predicting... \

00:34 Predicting... |

00:35 Predicting... /

00:35 Predicting... -

00:35 Predicting... \

00:35 Predicting... |

00:35 Predicting... /

00:36 Predicting... -

00:36 Predicting... \

00:36 Predicting... |

00:36 Predicting... /

00:36 Predicting... -

00:37 Predicting... \

00:37 Predicting... |

00:37 Predicting... /

00:37 Predicting... -

00:37 Predicting... \

00:38 Predicting... |

00:38 Predicting... /

00:38 Predicting... -

00:38 Predicting... \

00:38 Predicting... |

00:39 Predicting... /

00:39 Predicting... Done!


TabPFN-3 (thinking): AUC = 0.8672   (fit 487s)


## What just happened

One model, one switch, and the unclimbable wall is climbed: thinking mode does not just
close the TFM-vs-CatBoost gap on this dataset — it comes out ahead. On the full benchmark
protocol (mean over nine splits), the picture is: TabFM **0.858**, EXAONE-Tabular
**0.872**, CatBoost (tuned + ensembled) **0.882** — and thinking **0.885**, the only
single model above the CatBoost family on this dataset, trailing just the full AutoGluon
systems on the TabArena leaderboard.

The takeaways for practice:

- The metric argument matters here as everywhere (`thinking_metric="roc_auc"` — the lesson
  from notebook 02).
- Thinking mode trades fit-time compute for accuracy; reach for it when a dataset sits in
  a known TFM weak spot or when the last points of a metric are valuable. Details and
  options (`thinking_effort`, `thinking_timeout_s`) are in the
  [docs](https://docs.priorlabs.ai/capabilities/thinking-mode); the model itself is
  described in the [TabPFN-3 technical report](https://priorlabs.ai/technical-reports/tabpfn-3).
- It runs through the TabPFN API (`tabpfn-client`), not the local `tabpfn` package.

**Next**: [notebook 06](https://colab.research.google.com/github/Innixma/kdd2026_tutorial_materials/blob/main/notebooks/06_noniid_validation.ipynb) tackles data where rows aren't IID — honest validation with grouped and temporal splits.